### PR TITLE
Step 5 Datastore

### PR DESCRIPTION
**Pull request by andyp21**
- store comments in entities
- replace doPost Arraylist with Entities
Note: doGet is commented out because retrieval of entities is not covered in first part

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServletReverse.java [hunk 1]
```diff
@@ -20,32 +20,39 @@
...
+//     if (!history.isEmpty())
+//         response.getWriter().println(history.get(length));
+//   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Get the input from the form.
     String text = getParameter(request, "text-input", "");
     
+    Entity taskEntity = new Entity("Task");
```
Comment (735952) [inline portfolio/src/main/java/com/google/sps/servlets/DataServletReverse.java:44]: why not calling it comment?

File: portfolio/src/main/java/com/google/sps/servlets/DataServletDelete.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        return;
+    }
+    Filter keyFilter = new FilterPredicate("name", FilterOperator.EQUAL, name);
+    Query query = new Query("Entries").setFilter(keyFilter);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+
+    for (Entity entity : results.asIterable()) {
+            datastore.delete(entity.getKey());
```
Comment (ifranch) [inline portfolio/src/main/java/com/google/sps/servlets/DataServletDelete.java:66]: Format this line so it's properly tabulated

File: portfolio/src/main/java/com/google/sps/servlets/DataServletComments.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
+    // Get the input from the form.
+    String name = getParameter(request, "name", "Anonymous");
+    String text = getParameter(request, "comment", "");
+    long timestamp = System.currentTimeMillis();
+    UserService userService = UserServiceFactory.getUserService();
+    String id = userService.getCurrentUser().getUserId();
+
+    Entity users = new Entity("Entries",id);
```
Comment (ifranch) [inline portfolio/src/main/java/com/google/sps/servlets/DataServletComments.java:75]: Renaming users to userComment would make more sense

File: portfolio/src/main/java/com/google/sps/servlets/DataServletDelete.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.KeyRange;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+
+// Servlet which retrieves user String word entries, reverses them and returns the 2 most recent entries
```
Comment (ifranch) [inline portfolio/src/main/java/com/google/sps/servlets/DataServletDelete.java:36]: Remember to update the comment, here and also in the login and logout servlets

File: portfolio/src/main/java/com/google/sps/servlets/DataServletDelete.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+// Retrieves word entries and reverses them
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
+    // Get the user id
+    UserService userService = UserServiceFactory.getUserService();
+    String id = userService.getCurrentUser().getUserId();
+
+    // Filter the entity list for the clients id
+    Filter useridFilter = new FilterPredicate("id", FilterOperator.EQUAL, id);
```
Comment (ifranch) [inline portfolio/src/main/java/com/google/sps/servlets/DataServletDelete.java:55]: Proper camelCase is userIdFilter

**Pull request merged**